src/file-utils/edfi_lms_file_utils/file_reader.py

```diff
@@ -1,477 +1,478 @@
-    Reads the most recent grades files for all given sections into
-    a Pandas DataFrame.
-
-    Parameters
-    ----------
-    base_directory: str
-        The base / parent directory for LMS extractor files.
-    sections: pd.DataFrame
-        DataFrame containing sections read from a sections file.
-    nrows: int or None
-        (Optional) number of rows to read from the file - useful for testing
-        without reading the entirety of a large file.
-
-    Returns
-    -------
-    Pandas DataFrame with columns matching the model definition / CSV file.
-    """
-    return _get_data_for_section(base_directory, sections, get_grades, nrows)
-
-
-def get_attendance_events(
-    base_directory: str, section_id: int, nrows: Union[int, None] = None
-) -> pd.DataFrame:
-    """
-    Reads the most recent attendance events file for the given section into
-    a Pandas DataFrame.
-
-    Parameters
-    ----------
-    base_directory: str
-        The base / parent directory for LMS extractor files.
-    section_id: int
-        Section identifier.
-    nrows: int or None
-        (Optional) number of rows to read from the file - useful for testing
-        without reading the entirety of a large file.
-
-    Returns
-    -------
-    Pandas DataFrame with columns matching the model definition / CSV file.
-    """
-    file = fr.get_attendance_events_file(base_directory, section_id)
-
-    if file is not None:
-        return _read_csv(file, nrows)
-
-    return _default()
-
-
-def get_all_attendance_events(
-    base_directory: str, sections: pd.DataFrame, nrows: Union[int, None] = None
-) -> pd.DataFrame:
-    """
-    Reads the most recent attendance events files for all given sections into
-    a Pandas DataFrame.
-
-    Parameters
-    ----------
-    base_directory: str
-        The base / parent directory for LMS extractor files.
-    sections: pd.DataFrame
-        DataFrame containing sections read from a sections file.
-    nrows: int or None
-        (Optional) number of rows to read from the file - useful for testing
-        without reading the entirety of a large file.
-
-    Returns
-    -------
-    Pandas DataFrame with columns matching the model definition / CSV file.
-    """
-    return _get_data_for_section(base_directory, sections, get_attendance_events, nrows)
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from typing import Callable, Optional, Union
+
+import pandas as pd
+
+import edfi_lms_file_utils.file_repository as fr
+import edfi_lms_file_utils.constants as c
+
+logger = logging.getLogger(__name__)
+
+
+def _default() -> pd.DataFrame:
+    return pd.DataFrame()
+
+
+def _read_csv(file: str, nrows: Union[int, None] = None) -> pd.DataFrame:
+    logger.info(f"Reading file: {file}")
```

Comment:
Argh, the file endings changed! This line above is the only actual change. When I refactored the loader application to use this library, I realized that I had lost logging about what file is being read. This seemed like the best way to re-introduce that logging.